packages/core/src/server/watchFiles.ts

```diff
@@ -0,0 +1,56 @@
+import type { DevConfig, WatchFiles } from '@rsbuild/shared';
+import type { RsbuildDevMiddlewareOptions } from './getDevMiddlewares';
+
+export async function setupWatchFiles(
+  dev: DevConfig,
+  compileMiddlewareAPI: RsbuildDevMiddlewareOptions['compileMiddlewareAPI'],
+) {
+  const { watchFiles, hmr, liveReload } = dev;
+  if (!watchFiles || (!hmr && !liveReload)) {
+    return;
+  }
+
+  const watchFilesOptions = normalizeWatchFilesOptions(watchFiles);
+  if (!watchFilesOptions) {
+    return;
+  }
+
+  const chokidar = await import('@rsbuild/shared/chokidar');
+
+  const { paths, options } = watchFilesOptions;
+  const watcher = chokidar.watch(paths, options);
+
+  watcher.on('change', () => {
```

Comment:
I think we should close it when the server closes.

Reply:
Yeah, I should close it. [2b0bcbb](https://github.com/web-infra-dev/rsbuild/pull/2145/commits/2b0bcbbf81b38724aa0cad2339fc820f9163695d)